PROTOCOL.md

```diff
@@ -564,22 +565,60 @@ The following is an example `rowIdHighWaterMark` action:
 }
 ```
 
+### Domain Metadata
+The domain metadata action contains a configuration (string-string map) for a named metadata domain. Two overlapping transactions conflict if they both contain a domain metadata action for the same metadata domain.
+
+There are two types of metadata domains:
+1. **User-controlled metadata domains** have names that start with anything other than the `delta.` prefix. Any Delta client implementation or user application can modify these metadata domains, and can allow users to modify them arbitrarily. Delta clients and user applications are encouraged to use a naming convention designed to avoid conflicts with other clients' or users' metadata domains (e.g. `com.databricks.*` or `org.apache.*`).
+2. **System-controlled metadata domains** have names that start with the `delta.` prefix. This prefix is reserved for metadata domains mentioned in the Delta spec (i.e. as part of some table feature), and Delta client implementations must not allow users to modify the metadata for system-controlled domains. A Delta client implementation should only update metadata for system-controlled domains that it knows about and understands. System-controlled metadata domains are used by various table features and each table feature may impose additional semantics on the metadata domains it uses.
+
+The schema of the `domainMetadata` action is as follows:
+
+Field Name | Data Type | Description
+-|-|-
+domain | String | Identifier for this domain (system- or user-provided)
+configuration | Map[String, String] | A map containing configuration for the metadata domain
+removed | Boolean | When `true`, the action serves as a tombstone to logically delete a metadata domain
+
+Enablement:
+- The table must be on Writer Version 7.
+- A feature name `domainMetadata` must exist in the table's `writerFeatures`.
+
+#### Reader Requirements for Domain Metadata
+The reader should only read those domains they understand and ignore those it doesn't recognize.
```

Comment:
I think readers need to preserve all domains even if they don't understand them. We wouldn't expect the reader to understand specific user-controlled domains in the first place (user/app controls those), and system-controlled domains that need special reader attention are required to be part of some reader-writer table feature that can specify the desired behavior.

Reply:
Thanks, that makes sense. I reworded here based on the suggestion.